Snippets/ASP/ASTP_3/Behaviors.cs

```diff
@@ -76,4 +78,68 @@ public override async Task Invoke(IIncomingLogicalMessageContext context, Func<T
     }
 }
 
-#endregion
\ No newline at end of file
+#endregion
+
+#region BehaviorUsingIProvidePartitionKeyFromSagaId
+
+class OrderIdAsPartitionKeyBehavior : Behavior<IIncomingLogicalMessageContext>
+{
+    public OrderIdAsPartitionKeyBehavior(IProvidePartitionKeyFromSagaId partitionKeyFromSagaId)
+    {
+        partitionKeyFromSagaId1 = partitionKeyFromSagaId;
+    }
+
+    public override async Task Invoke(IIncomingLogicalMessageContext context, Func<Task> next)
+    {
+        var correlationProperty = SagaCorrelationProperty.None;
+        if (context.Message.Instance is IProvideOrderId provideOrderId)
+        {
+            var partitionKeyValue = provideOrderId.OrderId;
+            correlationProperty = new SagaCorrelationProperty("OrderId", partitionKeyValue);
+        }
+
+        await partitionKeyFromSagaId1.SetPartitionKey<OrderSagaData>(context, correlationProperty);
+
+        if (context.Headers.TryGetValue(Headers.SagaId, out var sagaIdHeader))
+        {
+            Log.Info($"Saga Id Header: {sagaIdHeader}");
```

Comment:
I was wondering if we should remove the log statements and just add comments into the snippet. Do you think they provide value on the configuration page?

Reply:
I'd keep it but as a Debug. If anyone is copying the code into production code at least they won't pay the penalty of logging it per message in production and they'll see it in the log when running with VS.

Reply:
Modified the logs to debug in the snippets for v3 and v4